core/src/main/java/com/novoda/noplayer/exoplayer/forwarder/MediaSourceOnErrorForwarder.java

```diff
@@ -1,32 +1,25 @@
-package com.novoda.noplayer.exoplayer;
+package com.novoda.noplayer.exoplayer.forwarder;
 
-import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
 import com.novoda.noplayer.Player;
 import com.novoda.noplayer.exoplayer.playererror.MediaSourceError;
 import com.novoda.noplayer.listeners.ErrorListeners;
 
 import java.io.IOException;
 
-class OnErrorForwarder extends ExoPlayerEventForwarder {
+class MediaSourceOnErrorForwarder implements ExtractorMediaSource.EventListener {
```

Comment:
Same as others above. We no longer extend from `ExoPlayerEventForwarder` but instead implement the interface we care about directly from `ExoPlayer`.